docs/rules/no-restricted-syntax.md

```diff
@@ -0,0 +1,51 @@
+# Disallow certain syntax (no-restricted-syntax)
+
+Some code bases prefer to not use certain features of JavaScript such as `FunctionExpression` or `WithStatement`. This rule warns on the passed node types.
```

Comment:
Everything wants me to write their docs :) 

Just think about it from the point of view of someone who is new to ESLint - what would they need to know to be able to use this rule? That's what should be here.